akeneo-design-system/src/components/Input/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,223 @@
+  const [isLastTagSelected, selectLastTag] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const containerRef = useRef<HTMLUListElement>(null);
+  const inputContainerRef = useRef<HTMLLIElement>(null);
+
+  const updateTags = useCallback(
+    (updatedTags: string[]) => {
+      updatedTags = updatedTags.slice(0, Math.min(MAX_TAGS, updatedTags.length));
+      setTags(updatedTags);
+    },
+    [setTags]
+  );
+
+  const onChangeCreateTags = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const tagsAsString = event.currentTarget.value;
+      if (tagsAsString !== '') {
+        let newTags = tagsAsString.split(/[\s,;]+/);
+        if (newTags.length === 1) {
+          return;
+        }
+        newTags = newTags.filter((tag: string) => tag.trim() !== '');
+
+        createTags([...tags, ...newTags]);
+      }
+    },
+    [tags]
+  );
+
+  const onBlurCreateTag = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const inputCurrentValue = event.currentTarget.value.trim();
+      if (inputCurrentValue !== '') {
+        createTags([...tags, ...[inputCurrentValue]]);
+      }
+    },
+    [tags]
+  );
+
+  const createTags = useCallback(
+    (newTags: string[]) => {
+      if (!allowDuplicates) {
+        newTags = arrayUnique(newTags);
+      }
+      updateTags(newTags);
+      if (inputRef && inputRef.current) {
+        inputRef.current.value = '';
+      }
+    },
+    [arrayUnique, inputRef, updateTags]
+  );
+
+  const removeTag = useCallback(
+    (tagIdToRemove: number) => {
+      const clonedTags = [...tags];
+      clonedTags.splice(tagIdToRemove, 1);
+      updateTags(clonedTags);
+    },
+    [tags, updateTags]
+  );
+
+  const focusOnInputField = useCallback(
+    (event: MouseEvent, ref: RefObject<HTMLElement>) => {
+      if (ref && ref.current === event.target && inputRef && inputRef.current) {
+        inputRef.current.focus();
+      }
+    },
+    [inputRef]
+  );
+
+  const handleTagDeletionUsingKeyboard = useCallback(
+    (event: KeyboardEvent) => {
+      if (![Key.Backspace.toString(), Key.Delete.toString()].includes(event.key)) {
+        selectLastTag(false);
+
+        if (tags.length >= MAX_TAGS) {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+
+        return;
+      }
+
+      if (tags.length === 0 || (inputRef && inputRef.current && inputRef.current.value.trim() !== '')) {
+        return;
+      }
+
+      if (!isLastTagSelected) {
+        selectLastTag(true);
+      } else {
+        const newTags = tags.slice(0, tags.length - 1);
+        updateTags(newTags);
+        selectLastTag(false);
+      }
+    },
+    [isLastTagSelected, selectLastTag, updateTags, tags, inputRef]
+  );
+
+  return (
+    <TagContainer
+      data-testid={'container'}
+      ref={containerRef}
+      isInvalid={isInvalid}
+      onClick={(event: MouseEvent) => focusOnInputField(event, containerRef)}
```

Comment:
```suggestion
      onClick={focusOnInputField}
```

as the containerRef is available in the component, I think it's better to not pass it